### PR TITLE
Override mathspeak with callback

### PR DESCRIPTION
**Pull request by nedredmond**
In order to internationalize the accessible labels, I have added a way to override the mathspeak functions on nodes with a callback that takes as its argument a `TeX` expression. We can use MathJax + SpeechRuleEngine to translate these into readable labels.

Issue: LC-1710

Testing:
Define the `setMathSpeakCallback` function on an EditableMathQuill instance, provide it a callback, and all mathspeak in aria-labels should be replaced with the result of the injected callback.

### CODE REVIEW

**Review by benchristel — COMMENTED** [review]
I don't think I have all the context here, but a couple ideas occurred to me as I read through this:

- The name `setMathSpeakCallback` confused me, because it defines the function  we use _instead_ of MathSpeak to format the math. Also, the word "callback" made me think it was going to be called asynchronously (e.g. when an event happens) and really, it's overriding the MathSpeak behavior. Maybe a name like `setSpeakableTextGenerator` would clarify the purpose of this method?
- It's slightly awkward that every subclass needs to know about the MathSpeak override function. I.e. this code is duplicated:
  ```ts
  if (this.mathSpeakCallback) {
    return this.mathSpeakCallback(this.latex());
  }
  ```
  Another solution would be to remove the implementation of `mathspeak` from the subclasses (letting it default to the superclass method) and let the superclass method look something like this:
  ```ts
  mathspeak() {
    if (this.mathSpeakCallback) {
      return this.mathSpeakCallback(this.latex());
    }
    builtInMathspeak();
  }
  ```
  Subclasses would override `builtInMathspeak` to implement their custom behavior.

**Comment by nedredmond** [comment]
@benchristel 

I can definitely change the name, and I agree about the duplication. I would like to avoid deleting the default behavior, but I can do like I did in the Superscript class:

```
if (this.mathSpeakCallback) {
  return super.mathspeak();
}
```

and direct funnel the behavior up to some common node. Not _all_ classes have the same behavior, but if they do with with `this.mathSpeakCallback(this.latex());`, we can direct them to a singular function in a parent class.

I was thinking of doing that anyway before merging, after fixing tests on Perseus' side.

**Comment by nedredmond** [comment]
> * The name `setMathSpeakCallback` confused me, because it defines the function  we use _instead_ of MathSpeak to format the math. Also, the word "callback" made me think it was going to be called asynchronously (e.g. when an event happens) and really, it's overriding the MathSpeak behavior. Maybe a name like `setSpeakableTextGenerator` would clarify the purpose of this method?

How about `setMathSpeakOverride`? We are, after all, just piping in MathSpeak from a different source.

**Comment by benchristel** [comment]
Sure, `setMathSpeakOverride` seems good to me.

Are you looking for an approval from me for this review, or did you tag me for awareness? I don't mean to block this PR from landing.

**Comment by nedredmond** [comment]
@benchristel you can approve or not, it's up to you! i made the suggested changes

**Review by benchristel — APPROVED** [review]
LGTM! I left a question inline about a regex.

File: src/commands/math.ts
Comment (benchristel) [inline src/commands/math.ts:324]: `/^(\\)+$/` will not match the empty string, so `mathspeakOverride` will be called if `tex` is empty. Is that okay? The comment  ("more than backslashes") isn't _quite_ accurate because of this edge case.

**Pull request merged**
